Pocket/pocket-ios
# feat(search): premium recent searches local cache

## Summary
Add local cache for recent saves for Premium Users. Tapping on the recent search does nothing currently.

## References 
IN-958, IN-989

## Implementation Details
Created `updateRecentSearches` method that contains the logic for the recent searches and uses user defaults to cache the array of strings. Created new SwiftUI view `RecentSearchView` and followed style structure similar to `Style+Settings.swift`.

## Test Steps
- [ ] WHEN “Saves” is selected
AND a premium user activates the search experience
THEN the search results area is empty of results
AND the search results area include the five-most recent searches as loaded from the local cache

- [ ] WHEN “Archive” is selected
AND a premium user activates the search experience
AND the device is online
THEN the search results area is empty of results
AND the search results area include the five-most recent searches as loaded from the local cache

- [ ] WHEN “All Items” is selected
AND a premium user activates the search experience
AND the device is online
THEN the search results area is empty of results
AND the search results area include the five-most recent searches as loaded from the local cache

- [ ] Given recent searches are to be displayed,
and there are no recent saves,
when the user views the search screen,
then the Recent Searches header will appear with a message along the lines of “Recent searches will appear here”.

- [ ] WHEN “Saves” is selected
AND a premium user activates the search experience
THEN the search results area is empty of results
AND the search results area include the five-most recent searches as loaded from the local cache

- [ ] When “Archive” is selected 
AND a premium user activates the search experience
AND the device is offline
THEN the search results area is empty of results
AND the search results area includes Offline View

- [ ] WHEN “All Items” is selected
AND a premium user activates the search experience
AND the device is offline
THEN the search results area is empty of results
AND the search results area includes Offline View

- [ ] Given recent searches are to be displayed,
and there are no recent saves,
when the user views the search screen,
then the Recent Searches header will appear with a message along the lines of “Recent searches will appear here”.

## PR Checklist:

## Screenshots
|Light Mode|Dark Mode|
|------|-----|
|![Light Mode](https://user-images.githubusercontent.com/6743397/204846998-a28c534d-2b87-4e46-9149-fbcb19280700.png)|![Dark Mode](https://user-images.githubusercontent.com/6743397/204847109-31ac4f26-c710-41a3-9978-70e9fb84bafa.png)|
